pkgs/development/libraries/libbsd/default.nix

```diff
@@ -1,18 +1,33 @@
-{ stdenv, fetchurl }:
+{ stdenv, fetchFromGitHub, autoreconfHook }:
 
 stdenv.mkDerivation rec {
   name = "libbsd-${version}";
   version = "0.8.3";
 
-  src = fetchurl {
-    url = "http://libbsd.freedesktop.org/releases/${name}.tar.xz";
-    sha256 = "1a1l7afchlvvj2zfi7ajcg26bbkh5i98y2v5h9j5p1px9m7n6jwk";
+  nativeBuildInputs = [ autoreconfHook ];
+
+  preAutoreconf = "mkdir m4";
+
+  patchPhase = ''
+    substituteInPlace configure.ac \
+      --replace "m4_esyscmd([./get-version])" "${version}"
+    sed -i '38i#undef strlcpy' include/bsd/string.h
+    sed -i '38i#undef strlcat' include/bsd/string.h
+    substituteInPlace src/setproctitle.c \
+     --replace 'extern typeof(setproctitle_impl) setproctitle_stub __attribute__((weak, alias("setproctitle_impl")));' ""
```

Comment:
ok I just combined everything into one patch. That way we won't change the hash for Linux systems.